internal/extproc/translator/gemini_helper.go

```diff
@@ -414,13 +414,22 @@ func openAIReqToGeminiGenerationConfig(openAIReq *openai.ChatCompletionRequest)
 		case openAIReq.ResponseFormat.OfJSONObject != nil:
 			gc.ResponseMIMEType = mimeTypeApplicationJSON
 		case openAIReq.ResponseFormat.OfJSONSchema != nil:
+			gc.ResponseMIMEType = mimeTypeApplicationJSON
 			var schemaMap map[string]any
 			if err := json.Unmarshal([]byte(openAIReq.ResponseFormat.OfJSONSchema.JSONSchema.Schema), &schemaMap); err != nil {
 				return nil, fmt.Errorf("invalid JSON schema: %w", err)
 			}
+			// it only works with gemini2.5 according to https://ai.google.dev/gemini-api/docs/structured-output#json-schema
+			if strings.Contains(openAIReq.Model, "gemini") && strings.Contains(openAIReq.Model, "2.5") { // TODO: users might be able to change the mapping?
```

Comment:
let's make sure the model name here is the one after the mutation using modelNameOverride field as the original model alias may not have the version name.